inference/conflict.go

```diff
@@ -17,16 +17,18 @@ package inference
 import (
 	"fmt"
 	"go/token"
+	"strings"
 
 	"go.uber.org/nilaway/annotation"
 	"go.uber.org/nilaway/util"
 	"golang.org/x/tools/go/analysis"
 )
 
 type conflict struct {
-	position token.Pos // stores position where the error should be reported
-	expr     string    // stores expression that is overcontrained (i.e., expected to be nonnil, but found nilable)
-	nilFlow  nilFlow   // stores nil flow from source to dereference point
+	position         token.Pos  // stores position where the error should be reported
+	expr             string     // stores expression that is overcontrained (i.e., expected to be nonnil, but found nilable)
+	nilFlow          nilFlow    // stores nil flow from source to dereference point
+	similarConflicts []conflict // stores other conflicts that are similar to this one
```

Comment:
Are we duplicating a lot of data by having this be an array of structs rather than pointers to structs? What's the Go idiomatic approach here?

Reply:
Good point. From what I know, there is not one governing Go idiomatic approach. It depends on the particular case. But I think for ours, we can definitely use a slice of struct pointers to save space and modify the object directly. Updated.